nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -7,6 +7,70 @@
 with lib;
 
 let
+  concat = lib.foldr (a: b: a+b) "";
+  /**
+   * Given a list of `options`, concats the result of mapping each options
+   * to a menuentry for use in grub.
+   *
+   *  * defaults: {name, image, params, initrd}
+   *  * options: [ option... ]
+   *  * option: {name, params, class}
+   */
+  menuBuilder_grub2 =
+  defaults: options: concat
+    (
+      map
+      (option: ''
+        menuentry '${defaults.name} ${
+        # Name appended to menuentry defaults to params if no specific name given.
+        if option ? name then
```

Comment:
Not entirely equivalent:
```
option.name or "(${option.params or ""})"
```
but this way of writing can be used to make the expression cleaner.

Reply:
I went for an in-between variation, using `option.name or` with the contents of the if after the `or`. Otherwise, menu options with default params went from "NixOS 18.03 Installer" to "NixOS 18.03 Installer ()".